pyiceberg/table/__init__.py

```diff
@@ -988,11 +988,15 @@ def commit_transaction(self) -> Table:
         Returns:
             The table with the updates applied.
         """
-        self._requirements = (AssertCreate(),)
-        self._table._do_commit(  # pylint: disable=W0212
-            updates=self._updates,
-            requirements=self._requirements,
-        )
+        if len(self._updates) > 0:
+            self._table._do_commit(  # pylint: disable=W0212
+                updates=self._updates,
+                requirements=(AssertCreate(),),
+            )
+
+        self._updates = ()
+        self._requirements = ()
```

Comment:
ah we cant use `super().commit_transaction()` here because the base Transaction class' `commit_transaction()` method adds a `AssertTableUUID(uuid=self.table_metadata.table_uuid),` as a requirement. 

Since requirements are checked before updates, this wont work. 

👍 im fine with leaving it as is. 
We just need to remember to clear `_updates` and `_requirements` for future subclasses of Transaction. And hopefully we'll follow the same pattern